aptos-move/framework/aptos-framework/sources/aptos_account.move

```diff
@@ -256,16 +256,16 @@ module aptos_framework::aptos_account {
         fungible_asset::is_address_balance_at_least(store_addr, amount)
     }
 
-    /// Burn from APT Primary FungibleStore
-    public(friend) fun burn_from_fungible_store(
+    /// Burn gas from APT Primary FungibleStore
+    public(friend) fun burn_from_fungible_store_for_gas(
```

Comment:
why are we naming these explicitly but there's no limitation to how they are used?

Reply:
as long as they are public(friend) and only used for gas - that is the limitation